source/MicroBitPowerManager.cpp

```diff
@@ -762,25 +762,6 @@ int MicroBitPowerManager::canDeepSleep( bool wakeOnTime, CODAL_TIMESTAMP wakeUpT
             return DEVICE_INVALID_STATE;
         }
     }
-    else
-    {
-        if ( wakeUpSources)
-        {
-            deepSleepCallbackData data;
-            CodalComponent::deepSleepAll( deepSleepCallbackCountWakeUps, &data);
-            if ( data.count == 0)
-            {
-                return DEVICE_INVALID_STATE;
-            }
-        }
-        else
-        {
-            if ( wakeUpPin == NULL)
-            {
-                return DEVICE_INVALID_STATE;
-            }
-        }
-    }
```

Comment:
Looks fine to me - I think the behaviour of deepsleeping when asked, even if there are no wake up sources, is a good behaviour.